controllers/horizon_controller.go

```diff
@@ -547,3 +579,52 @@ func (r *HorizonReconciler) ensureHorizonSecret(
 
 	return nil
 }
+
+func (r *HorizonReconciler) renderMemcached(instance *horizonv1alpha1.Horizon) *memcachedv1.Memcached {
+	return &memcachedv1.Memcached{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "memcached.openstack.org/v1beta1",
+			Kind:       "Memcached",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-memcached", instance.Name),
+			Namespace: instance.Namespace,
+		},
+		Spec: memcachedv1.MemcachedSpec{
+			Replicas: instance.Spec.Replicas,
+		},
+	}
+}
+
+func (r *HorizonReconciler) getMemcachedCR(ctx context.Context, memcachedName string) (bool, error) {
+	list := &memcachedv1.MemcachedList{}
+	if err := r.Client.List(ctx, list); err != nil {
+		r.Log.Info("Error while getting list of memcaches")
+		return false, err
+	}
+	for _, memcached := range list.Items {
+		if memcached.ObjectMeta.Name == memcachedName {
+			r.Log.Info("Found Memcached CR for Horizon!")
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (r *HorizonReconciler) checkMemcachedSvc(memcachedName string, h *helper.Helper) (svc string, err error) {
+	labelSelector := k8s_labels.Set{"app": "memcached"}.String()
+	svcList, err := h.GetKClient().CoreV1().Services("").List(context.TODO(), metav1.ListOptions{
```

Comment:
I think we have a lib-common helper for this https://github.com/openstack-k8s-operators/lib-common/blob/master/modules/common/service/service.go#L218

Reply:
Implemented this in the latest update too. Thanks for the pointer.